plugins/uf2/fu-uf2-device.c

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-uf2-device.h"
+#include "fu-uf2-firmware.h"
+
+struct _FuUf2Device {
+	FuUdevDevice parent_instance;
+	guint64 family_id;
+};
+
+G_DEFINE_TYPE(FuUf2Device, fu_uf2_device, FU_TYPE_UDEV_DEVICE)
+
+static FuFirmware *
+fu_uf2_device_prepare_firmware(FuDevice *device,
+			       GBytes *fw,
+			       FwupdInstallFlags flags,
+			       GError **error)
+{
+	FuUf2Device *self = FU_UF2_DEVICE(device);
+	g_autoptr(FuFirmware) firmware = fu_uf2_firmware_new();
+
+	if (!fu_firmware_parse(firmware, fw, flags, error))
+		return NULL;
+
+	/* check the family_id matches if we can read the old firmware */
+	if (self->family_id > 0 && fu_firmware_get_idx(firmware) > 0 &&
+	    self->family_id != fu_firmware_get_idx(firmware)) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_NOT_SUPPORTED,
+			    "family ID was different, expected 0x%08x bytes and got 0x%08x",
```

Comment:
family id doesn't have bytes, isn't it just a number?

Reply:
Oops, thanks.